### PR TITLE
Adding Netlify settings through `netlify.toml`

### PR DESCRIPTION
**Pull request by AnthonyJCampbell**
# Description

Adding a `netlify.toml` file to create context-specific environmental variables for the deployment on the front-end on Netlify.

## Checklist

- [x] I have made corresponding changes to the documentation
- [ ] Deployment works as expected, without any errors.

### CODE REVIEW

**Comment by AnthonyJCampbell** [comment]
[Sort out Environment-specific URLs for API access](https://trello.com/c/OrHMn5r5/52-sort-out-environment-specific-urls-for-api-access)

File: README.md [hunk 1]
```diff
@@ -115,11 +115,14 @@ Volunteering your time; it pays you and your whole community fantastic dividends
...
-In order for the app to function correctly, the user must set up their own environment variables. There should be a .env file containing the following:
+In order for the app to function correctly, the user need not set up their own environment variables at this point in time. However, inspection of the `start` command will reveal that it injects `REACT_APP_API_URL=http://localhost:9000` when starting the app on a local machine (during development). This points to the port the back-end server will run on when ran on the local machine. This is done to pre-seed the environment.
+
+The exact API_URL that's used is dependent on the environment it runs in. As mentioned above, if the app is run on one's local machine, it will look for `http://localhost:9000`. However, on Netlify, it'll target different API URLs, depending on if it's in the deploy-preview or production environment.
+
 
 🚫These are just examples, replace them with the specifics for your app
 
-    *  REACT_APP_apiKey - this is your Google API key, which can be generated in the Google Cloud Console
+    *  REACT_APP_API_URL - this is your Google API key, which can be generated in the Google Cloud Console
```
Comment (Pav0l) [inline README.md]: I have a feeling the `REACT_APP_API_URL` is not our Google API key 😂

File: src/index.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ import ReactDOM from 'react-dom';
 import App from './App';
 import * as serviceWorker from './utils/serviceWorker';
 
+require('dotenv').config()
```
Comment (Pav0l) [inline src/index.js]: Not sure if we still need this. 🐛

File: src/utils/generalVariables.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-/**
- * @TODO Add environment variable with URL based on environment
- */
-const URL = 'https://tieme-ndo-backend-staging1.herokuapp.com/api/v1';
+// Retrieving URL from environment - compatible with deployment context
+// When running on a local machine, the URL will default to REACT_APP_API_URL - provided the app is run using `yarn start`
+// When hosted on Netlify, it'll retrieve the requisite environmental URL and target the appropriate API
+export const URL = process.env.API_URL || process.env.REACT_APP_API_URL;
```
Comment (Pav0l) [inline src/utils/generalVariables.js]: Great explanation if the comment 💯

File: src/components/pages/Login.js [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { useState } from "react";
 import { Redirect } from "react-router-dom";
 import { loginHandler } from "../../handlers/authenticationHandlers.js";
 
+import { URL } from './../../utils/generalVariables'
```
Comment (Pav0l) [inline src/components/pages/Login.js]: 💡 Not sure here, but I though we'd be using `URL` in the "path" variable for each request handler. And then we just import that "path" variable. 

For example:
```
paths = {
    login: ${URL}/login
}
...
import { login} ...

...

axios.post(login, ...)
```
But I might be wrong here.

File: README.md (hunk 1 above)
Comment (AnthonyJCampbell) [reply README.md]: Whoops. Kept the placeholder 😂

File: src/index.js (hunk 2 above)
Comment (AnthonyJCampbell) [reply src/index.js]: Good catch! We don't will remove.

File: src/utils/generalVariables.js (hunk 3 above)
Comment (AnthonyJCampbell) [reply src/utils/generalVariables.js]: 🙏

File: src/components/pages/Login.js (hunk 4 above)
Comment (AnthonyJCampbell) [reply src/components/pages/Login.js]: You are correct. It's an import I forgot to remove after some testing and tweaking. Will remove!

**Pull request merged**